### PR TITLE
AMQNET-619: Handle remote detach properly for sender link

### PR DESCRIPTION
**Pull request by Havret**

### CODE REVIEW

File: src/NMS.AMQP/Provider/Amqp/AmqpProducer.cs [hunk 1]
```diff
@@ -53,13 +53,13 @@ public Task Attach()
             };
 
             string linkName = info.Id + ":" + target.Address;
-            var taskCompletionSource = new TaskCompletionSource<bool>();
-            senderLink = new SenderLink(session.UnderlyingSession, linkName, frame, (link, attach) => { taskCompletionSource.SetResult(true); });
+            var tsc = new TaskCompletionSource<bool>();
```
Comment (michaelandrepearce) [inline src/NMS.AMQP/Provider/Amqp/AmqpProducer.cs]: Can you revert the shortening of the var. Unneeded change

Comment (Havret) [reply src/NMS.AMQP/Provider/Amqp/AmqpProducer.cs]: Sure thing. Fixed.

File: src/NMS.AMQP/Provider/Amqp/AmqpProducer.cs [hunk 2]
```diff
@@ -53,13 +53,13 @@ public Task Attach()
             };
 
             string linkName = info.Id + ":" + target.Address;
-            var taskCompletionSource = new TaskCompletionSource<bool>();
-            senderLink = new SenderLink(session.UnderlyingSession, linkName, frame, (link, attach) => { taskCompletionSource.SetResult(true); });
+            TaskCompletionSource<bool> tsc = new TaskCompletionSource<bool>();
```
Comment (michaelandrepearce) [inline src/NMS.AMQP/Provider/Amqp/AmqpProducer.cs]: You made further changes...:s the name is still shortened  i meant to put it back to var taskCompletionService so there is no change in this line.

Comment (Havret) [reply src/NMS.AMQP/Provider/Amqp/AmqpProducer.cs]: Hope it's ok now. 😳

**Pull request merged**